elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -1006,7 +1007,49 @@ describe('peerPool', () => {
 		});
 	});
 
-	describe.skip('#_evictPeer', () => {});
+	describe('#_evictPeer', () => {
+		const whitelistedPeers = [
+			{ peerId: '1.2.3.4:5000', ipAddress: '1.2.3.4', wsPort: 5000 },
+		];
+		const fixedPeers = [
+			{ peerId: '5.6.7.8:5000', ipAddress: '5.6.7.8', wsPort: 5000 },
+		];
+		const defaultPeers = [
+			{
+				id: '69.123.456.78:5000',
+				peerId: '69.123.456.78:5000',
+				ipAddress: '69.123.456.78',
+				wsPort: 5000,
+			},
+			...whitelistedPeers.map(peer => ({ ...peer, id: peer.peerId })),
+			...fixedPeers.map(peer => ({ ...peer, id: peer.peerId })),
+		];
+
+		beforeEach(async () => {
+			(peerPool as any)._peerLists.whitelisted = whitelistedPeers;
+			(peerPool as any)._peerLists.fixedPeers = fixedPeers;
+			sandbox.stub(peerPool as any, 'getPeers').returns(defaultPeers);
+			sandbox.stub(peerPool, 'removePeer');
+		});
+
+		it('should not evict whitelisted peer', async () => {
+			(peerPool as any)._evictPeer(InboundPeer);
+			expect(peerPool.removePeer).not.to.be.calledWith(
+				whitelistedPeers[0].ipAddress,
+				sinon.match.any,
+				sinon.match.any,
+			);
+		});
+
+		it('should not evict fixed peer', async () => {
+			(peerPool as any)._evictPeer(InboundPeer);
+			expect(peerPool.removePeer).not.to.be.calledWith(
+				fixedPeers[0].ipAddress,
+				sinon.match.any,
```

Comment:
We're not using this anywhere else. Is it necessary?

Reply:
for this test case, the second and third parameter is not important, so that's why it's using any

Reply:
You can also test it without because you're not utilising `calledWithExactly`.

Reply:
Updated to use called with exactly 😉